src/core/qgsziputils.cpp

```diff
@@ -85,6 +85,14 @@ bool QgsZipUtils::unzip( const QString &zipFilename, const QString &dir, QString
         if ( zip_fread( file, buf, len ) != -1 )
         {
           QFileInfo newFile( QDir( dir ), QString( stat.name ) );
+          QString fileName( stat.name );
+          bool isDirectory = fileName.lastIndexOf( "/" ) == fileName.length() - 1;
+
+          if ( isDirectory )
+          {
+            QDir( dir ).mkdir( fileName );
+          }
+
```

Comment:
writing of file with std::ofstream should be done only when we do not have a directory...